scripts/quests/sandoria/DRG_AF2_Chasing_Quotas.lua

```diff
@@ -0,0 +1,272 @@
+                player:getQuestStatus(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.A_CRAFTSMAN_S_WORK) == QUEST_COMPLETED
+        end,
+
+        [xi.zone.PORT_SAN_DORIA] =
+        {
+            ['Arminibit'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getVar("Quest[0][95]csSeen") == 0 then
+                        return quest:progressEvent(18)
+                    elseif player:getVar("Quest[0][95]csSeen") == 1 then --Declined First Offering.
+                        return quest:progressEvent(14)
+                    end
+                end,
+            },
+
+            ['Ceraulian'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getVar("Quest[0][95]csSeen") == 0 then
+                        return quest:progressEvent(18)
+                    elseif player:getVar("Quest[0][95]csSeen") == 1 then --Declined First Offering.
+                        return quest:progressEvent(14)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [14] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        quest:setVar(player, 'csSeen', 0)
+                    else
+                        quest:setVar(player, 'csSeen', 1) -- Declined First Offering.
+                    end
+                end,
+
+                [18] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'csSeen', 0)
+                    else
+                        quest:setVar(player, 'csSeen', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_SAN_DORIA] =
+        {
+            ['Ceraulian'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        if player:getCharVar("Timer") > os.time() then
+                            return quest:progressEvent(3)
+                        else
+                            return quest:progressEvent(7)
+                        end
+                    elseif questProgress == 2 then
+                        return quest:progressEvent(8)
+                    elseif questProgress == 3 then
+                        return quest:progressEvent(6)
+                    elseif questProgress == 4 or questProgress == 5 then
+                        return quest:progressEvent(9)
+                    elseif questProgress == 6 then
+                        return quest:progressEvent(15)
```

Comment:
Another thing to note:

We reguarly reserve `:progressEvent` for the important events that actually progress the quest and use `:event` for the temporal dialog changes that don't actually do anything/aren't important.

The reson for this is that this npc may, or may not, be involved in other quests/missions.

`:progressEvent` will ALWAYS take priority over `:event` meaning we will never "block" any other quest or mission progression. Not really retail accurate, since retail DOES block progression in certain cases, but this falls into "be better than retail, let's not randomly make players thing they did something wrong when they didnt"

Reply:
If it needs an override because it's important van use event():importantOnce()

Reply:
Indeed, i usually mix up a couple - if it has an eventFinish then its a progressEvent otherwise, tis event.

Thanks for the feedback - made the appropriate changes.